sdk/tables/azure-data-tables/src/serializers.cpp

```diff
@@ -527,7 +527,18 @@ namespace Azure { namespace Data { namespace Tables { namespace _detail {
   {
     Models::TableEntity tableEntity{};
 
-    auto properties = json.get<std::map<std::string, std::string>>();
+    std::map<std::string, std::string> properties;
+    for (auto it = json.items().begin(); it != json.items().end(); ++it) {
+      const std::string& key = it.key();
+      const auto& value = it.value();
+      if (value.is_string()) {
+          properties[key] = value.get<std::string>();
+      } 
+      else {
+          properties[key] = value.dump();
+      }
+    }

```

Comment:
Similar to https://github.com/Azure/azure-sdk-for-cpp/pull/5774#issuecomment-2221651154, can we add a unit test which exercises this code path?

Reply:
@gearama please take a look when you get a chance.

Reply:
> can we add a unit test which exercises this code path?

let me see...

Reply:
Added the unit test to hit that code